src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -25,6 +25,42 @@ namespace ArborX
 namespace Details
 {
 
+#ifdef ARBORX_ENABLE_MPI
+// FIXME: Used only by the DistributedTree. Should be removed once proper
+// callback abstractions are in place.
+template <typename BVH, typename ReversePermutation>
+KOKKOS_FUNCTION typename BVH::bounding_volume_type const &
+getPrimitiveBoundingVolume(BVH const &bvh,
+                           ReversePermutation const &rev_permute,
+                           int leaf_index)
+{
+  int const leaf_nodes_shift = bvh.size() - 1;
+  return bvh.getBoundingVolume(
+      bvh.getNodePtr(rev_permute(leaf_index) + leaf_nodes_shift));
+}
+
+template <typename ExecutionSpace, typename BVH>
+Kokkos::View<unsigned int *, typename BVH::memory_space>
+getReversePermutation(ExecutionSpace const &exec_space, BVH const &bvh)
+{
```

Comment:
Not a fan of where this functions leave. They are only used by the distributed tree. But, in order to make functions friends to `BVH`, their declaration (or definition) must be available prior to that, which is not possible with distributed.